app/src/main/java/com/onboarding/parkingsystemjava/database/ParkingDatabase.java

```diff
@@ -41,4 +44,31 @@ public int getParkingLots() {
     public void setParkingLots(int parkingLots) {
         this.parkingLots = parkingLots;
     }
+
+    public int updateDatabase() {
+        Calendar currentTime = new GregorianCalendar();
+        HashMap<Integer, ArrayList<Reservation>> reservations = this.getReservations();
+        int deleted = ConstantUtils.INT_ZERO;
+        for (int i = ConstantUtils.INT_ZERO; i <= this.parkingLots; i++) {
+            if (reservations.containsKey(i)) {
+                deleted += searchLotForOldReservation(currentTime, this.reservations.get(i));
+                if (reservations.get(i).isEmpty()) {
+                    reservations.remove(i);
+                }
+            }
+        }
+        return deleted;
+    }
+
+    private int searchLotForOldReservation(Calendar current, ArrayList<Reservation> reservations) {
```

Comment:
i think deleteOldReservations...(or something like that)its more representative

Reply:
done